### PR TITLE
gh-121103: Put free-threaded libraries in `lib/python3.14t`

### PR DESCRIPTION
**Pull request by colesbury**
On POSIX systems, excluding macOS framework installs, the lib directory for the free-threaded build now includes a "t" suffix to avoid conflicts with a co-located default build installation.

* Issue: gh-121103

### CODE REVIEW

**Comment by colesbury** [comment]
`pip`, `uv`, `venv`, `setuptools`, seem to work, but I don't entirely know what I'm doing, so I'd appreciate a thorough review.

**Review by corona10 — APPROVED** [review]

**Comment by hroncok** [comment]
Does this mean that all of the pure Puythomn standard library files would need to be installed in lib/python3.14t
 as well? I.e. when I have both the "normal" and the freethreaded Python installed, do I need all the standard library modules twice?

**Comment by encukou** [comment]
I don't have very deep knowledge in this area, so I can't easily do a thorough review :(
I'd trust @FFY00 and @hroncok's reviews if they can contribute their time for it.

**Comment by erlend-aasland** [comment]
> Does this mean that all of the pure Python standard library files would need to be installed in lib/python3.14t as well? I.e. when I have both the "normal" and the freethreaded Python installed, do I need all the standard library modules twice?

As far as I can tell, from a quick look at this PR: yes.

**Comment by hroncok** [comment]
> As far as I can tell, from a quick look at this PR: yes.

I thought so.

The real question is: Is this the outcome we anticipated?

If not, I suggest to let the stdlib and platstdlib values intact.

**Comment by colesbury** [comment]
Yes, that was the intention. The goal is to separate the installation. We run into a lot of potential problems if we have partially overlapping stdlib files.

**Comment by erlend-aasland** [comment]
> Yes, that was the intention. The goal is to separate the installation. We run into a lot of potential problems if we have partially overlapping stdlib files.

FTR, I totally agree with this.

**Comment by colesbury** [comment]
To expand on my previous comment: if we have kept a common stdlib and platstdlib, the Python files would be shared, but the `.so` files would not. So I'd be concerned that if someone installed, e.g. 3.13.0 default and 3.13.8 free-threaded to the same prefix, then the .py files and native .so libraries might not match. Keeping things separate in `lib/python3.13t` vs. `lib/python3.13` avoids this problem.

**Comment by colesbury** [comment]
I plan to merge this tomorrow if there are no objections

File: Makefile.pre.in [hunk 1]
```diff
@@ -41,6 +41,7 @@ AR=		@AR@
 READELF=	@READELF@
 SOABI=		@SOABI@
 ABIFLAGS=	@ABIFLAGS@
+ABI_THREAD= @ABI_THREAD@
```
Comment (ned-deily) [inline Makefile.pre.in]: Nit: the Makefile uses tabs here rather than spaces. s/= /=^T/

**Review by ned-deily — APPROVED** [review]
Other than the nit comment, +1. This looks like it does what it promises. We still have a bit more to do before a free-threaded build can be installed to the same prefix as a non-free-threaded build without conflicts, primarily due to duplicate symlinked names in the `bin` and `lib/pkgconfig` directories. But this isn't really a new problem as there are similar conflicts when sharing with a debug build prior to this PR. I'd say let's merge this PR now and we can follow up in another PR for which I want to do some more investigation.

**Review by erlend-aasland — APPROVED** [review]

File: Lib/site.py [hunk 2]
```diff
@@ -390,14 +394,18 @@ def getsitepackages(prefixes=None):
...
+            abi_thread = ''
         if os.sep == '/':
             libdirs = [sys.platlibdir]
             if sys.platlibdir != "lib":
                 libdirs.append("lib")
 
             for libdir in libdirs:
                 path = os.path.join(prefix, libdir,
-                                    f"{implementation}{ver[0]}.{ver[1]}",
+                                    f"{implementation}{ver[0]}.{ver[1]}{abi_thread}",
```
Comment (AlexWaygood) [inline Lib/site.py:408]: @colesbury -- would you be able to update the [docs for the `site` module](https://docs.python.org/3.13/library/site.html) to reflect these changes to how Python finds the `site-packages` directory?

(The context is I'm currently working on reimplementing Python's logic here for Astral's work-in-progress type checker. I was reading through the `site.py` source code, and it surprised me to find logic here that wasn't documented!)

Comment (colesbury) [reply Lib/site.py:408]: Thanks for the pointer. I put up a PR to update the docs: https://github.com/python/cpython/pull/122737

**Pull request merged**